### ISSUE TITLE
[restrict-plus-operands] False positive when importing type

### ISSUE THREAD

**Issue by gogakoreli** [typescript-eslint/typescript-eslint#3221]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

```JSON
{
  "rules": {
    "@typescript-eslint/restrict-plus-operands": "error"
  }
}
```

```TS
// main.ts
import { Person } from './person'
const person: Person = { name: '' };
const name = person.name + ''; // restrict-plus-operands eslint error

// person.ts
export interface Person {
  name: string;
}
```

**Expected Result**
Shouldn't be an error

**Actual Result**
Operands of '+' operation must either be both strings or both numbers. Consider using a template literal. eslint(@typescript-eslint/restrict-plus-operands)

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.18.0` |
| `@typescript-eslint/parser`        | `4.18.0` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `7.22.0` |
| `node`                             | `14.16.0` |

**Comment by bradzacher** [comment]
All our rules support imported types.
If this isn't working for you - then likely you have a problem with your configuration, or something has gotten out of sync.

**Comment by gogakoreli** [comment]
Thanks for guiding me to look the problem in the configuration, that was in fact the case in my project. To my surprise linter was working well enough for me to doubt that. In my case this issue has to do with parserOptions and tsconfig configuration.
